### PR TITLE
Drop interim RaftLogWriter interface and inline it via RaftLog

### LINKED ISSUE #6304: Drop RaftLogWriter and use RaftLog directly everywhere
**Description**

As a third step for #6147, and in preparation for the `RaftLog` to directly support appending, we will drop the `RaftLogWriter` interface completely but keep its method in `RaftLog` - the only difference is all callers will now call them via the `RaftLog` type.

Blocked by #6302

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR drops the interim `RaftLogWriter` interface and inlines it in `RaftLog`, with all callers now calling the `RaftLog` directly. Some tests were adapted, but no new tests were written yet for `RaftLog` as we expect this API to be short lived.

## Related issues

closes #6304 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: atomix/cluster/src/main/java/io/atomix/raft/storage/log/RaftLogWriter.java [hunk 1]
```diff
@@ -17,13 +16,6 @@
...
+import io.atomix.storage.journal.JournalWriter;
 
-/** Raft log writer. */
-public class RaftLogWriter extends DelegatingJournalWriter<RaftLogEntry> {
-
-  public RaftLogWriter(final SegmentedJournalWriter<RaftLogEntry> writer) {
-    super(writer);
-  }
-}
+public interface RaftLogWriter extends JournalWriter<RaftLogEntry> {}
```
Comment (deepthidevaki) [inline atomix/cluster/src/main/java/io/atomix/raft/storage/log/RaftLogWriter.java:21]: This interface is not used anymore.

**Review by deepthidevaki — CHANGES_REQUESTED** [review]
:+1:  You can delete RaftLogWriter interface and then it is good to merge.

File: atomix/cluster/src/main/java/io/atomix/raft/storage/log/RaftLogWriter.java (hunk 1 above)
Comment (npepinpe) [reply atomix/cluster/src/main/java/io/atomix/raft/storage/log/RaftLogWriter.java]: :see_no_evil: That was the whole point of the PR! Sorry haha

**Review by deepthidevaki — APPROVED** [review]
:tada:

**Pull request merged**